libs/pipelines/api_pipeline.py

```diff
@@ -67,10 +72,34 @@ def run_on_all_fips_for_intervention(
     return all_timeseries
 
 
+def generate_metrics_and_latest_for_fips(
+    fips,
+) -> [List[MetricsTimeseriesRow], Optional[MetricsTimeseriesRow]]:
+    """
+    For a FIPS, generate a MetricsTimeseriesRow per day and return the latest.
+
+    Args:
+        fips: FIPS to run on.
+
+    Returns:
+        Tuple of MetricsTimeseriesRows for all days and the latest.
+    """
+    metrics_timeseries = (
+        calculate_top_level_metrics_for_fips(fips).reset_index().to_dict(orient="records")
+    )
+    metrics_for_fips = [MetricsTimeseriesRow(**metric_row) for metric_row in metrics_timeseries]
+    if len(metrics_for_fips):
+        latest = metrics_for_fips[-1]
+    else:
+        latest = None
+    return metrics_for_fips, latest
+
+
 def build_timeseries_for_fips(
     intervention, us_latest, us_timeseries, model_output_dir, fips
 ) -> Optional[RegionSummaryWithTimeseries]:
     fips_latest = us_latest.get_record_for_fips(fips)
+    metrics_timeseries, metrics_latest = generate_metrics_and_latest_for_fips(fips)
```

Comment:
since you have the timeseries and latest datasets for a given fips in this function, I might recommend calling the version of the function `calculate_top_level_metrics_for_timeseries` so that you're not filtering by fips twice.